### PR TITLE
Remove OBS_UNUSED

### PR DESCRIPTION
**Pull request by tytan652**
### Description

`OBS_UNUSED` is not portable to MSVC and so will not prevent it from emitting error once we switch to /W4.

### Motivation and Context

Just wanted to do it.

### How Has This Been Tested?

I build with `-Werror=unused-parameter` and suppress FTLs warning. No error because of unused parameter.

### Types of changes

- Code cleanup (non-breaking change which makes code smaller or more readable)

### Checklist:

- [x] My code has been run through [clang-format](https://github.com/obsproject/obs-studio/blob/master/.clang-format).
- [x] I have read the [**contributing** document](https://github.com/obsproject/obs-studio/blob/master/CONTRIBUTING.rst).
- [x] My code is not on the master branch.
- [x] The code has been tested.
- [x] All commit messages are properly formatted and commits squashed where appropriate.
- [x] I have included updates to all appropriate documentation.

### CODE REVIEW

**Review by RytoEX — APPROVED** [review]
Looks good to me. @jpark37 @PatTheMav are these changes now acceptable?

File: libobs/graphics/graphics-ffmpeg.c [hunk 1]
```diff
@@ -152,10 +152,12 @@ static void *ffmpeg_image_copy_data_straight(struct ffmpeg_image *info,
 	return data;
 }
 
-static inline size_t get_dst_position(size_t src OBS_UNUSED, const size_t w,
+static inline size_t get_dst_position(size_t src, const size_t w,
```
Comment (jpark37) [inline libobs/graphics/graphics-ffmpeg.c:155]: Remove src parameter entirely?

File: libobs/obs-scene.c [hunk 2]
```diff
@@ -2060,8 +2064,10 @@ static inline bool source_has_audio(obs_source_t *source)
 static obs_sceneitem_t *obs_scene_add_internal(obs_scene_t *scene,
 					       obs_source_t *source,
 					       obs_sceneitem_t *insert_after,
-					       bool create_texture OBS_UNUSED)
+					       bool create_texture)
```
Comment (jpark37) [inline libobs/obs-scene.c:2067]: Remove parameter.

File: libobs/obs-video.c [hunk 3]
```diff
@@ -599,11 +599,13 @@ static const uint8_t *set_gpu_converted_plane(uint32_t width, uint32_t height,
 	return in;
 }
 
-static void set_gpu_converted_data(struct obs_core_video *video OBS_UNUSED,
+static void set_gpu_converted_data(struct obs_core_video *video,
```
Comment (jpark37) [inline libobs/obs-video.c:602]: Remove parameter.

File: libobs/graphics/graphics-ffmpeg.c (hunk 1 above)
Comment (RytoEX) [reply libobs/graphics/graphics-ffmpeg.c:155]: It does seem as though this was left in after adjustments on the initial PR, but it is not used at all in the function in its current form. I would agree that it's fine to remove.

File: plugins/aja/aja-common.cpp [hunk 4]
```diff
@@ -201,8 +201,7 @@ void populate_pixel_format_list(NTV2DeviceID deviceID, obs_property_t *list)
 	}
 }
 
-void populate_sdi_transport_list(obs_property_t *list,
-				 IOSelection io OBS_UNUSED,
+void populate_sdi_transport_list(obs_property_t *list, IOSelection,
```
Comment (jpark37) [inline plugins/aja/aja-common.cpp:204]: Remove parameter?

File: plugins/aja/aja-common.cpp [hunk 5]
```diff
@@ -1019,7 +1018,7 @@ inline bool IsStandard1080p(NTV2Standard standard)
 	return false;
 }
 
-VPIDStandard DetermineVPIDStandard(NTV2DeviceID id OBS_UNUSED, IOSelection io,
+VPIDStandard DetermineVPIDStandard(NTV2DeviceID, IOSelection io,
```
Comment (jpark37) [inline plugins/aja/aja-common.cpp:1021]: Remove parameter?

**Review by jpark37 — CHANGES_REQUESTED** [review]
I've gone through the changes, and marked the ones I think can be removed, and not just tagged.

File: UI/frontend-plugins/aja-output-ui/aja-ui-main.cpp [hunk 6]
```diff
@@ -451,8 +450,10 @@ static void OBSEvent(enum obs_frontend_event event, void *)
 	}
 }
 
-static void aja_loaded(void *data OBS_UNUSED, calldata_t *calldata)
+static void aja_loaded(void *data, calldata_t *calldata)
```
Comment (jpark37) [inline UI/frontend-plugins/aja-output-ui/aja-ui-main.cpp:453]: C++ file. Just remove var name..

Comment (tytan652) [reply UI/frontend-plugins/aja-output-ui/aja-ui-main.cpp:453]: Unlike the other one, it's a static function (and callback) so there is no function declaration to get the name of the parameter, this is why I didn't just remove the name.

Comment (jpark37) [reply UI/frontend-plugins/aja-output-ui/aja-ui-main.cpp:453]: How about `void */* data */`? (Or whatever clang-format refactors that to.)

Comment (tytan652) [reply UI/frontend-plugins/aja-output-ui/aja-ui-main.cpp]: Should be good now

**Review by jpark37 — APPROVED** [review]

**Pull request merged**